dubbo-rpc/dubbo-rpc-injvm/src/main/java/org/apache/dubbo/rpc/protocol/injvm/InjvmInvoker.java

```diff
@@ -67,6 +79,36 @@ public Result doInvoke(Invocation invocation) throws Throwable {
         if (serverHasToken) {
             invocation.setAttachment(Constants.TOKEN_KEY, serverURL.getParameter(Constants.TOKEN_KEY));
         }
-        return exporter.getInvoker().invoke(invocation);
+
+        if (isAsync(exporter.getInvoker().getUrl(), getUrl())) {
+            ((RpcInvocation) invocation).setInvokeMode(InvokeMode.ASYNC);
+            // use consumer executor
+            ExecutorService executor = executorRepository.getExecutor(getUrl());

```

Comment:
how about directly use `createExecutorIfAbsent`

Reply:
Yes, that's batter.